src/analysis/retail/deathknight/unholy/modules/talents/CommanderOfTheDead.tsx

```diff
@@ -0,0 +1,127 @@
+import { t } from '@lingui/macro';
+import SPELLS from 'common/SPELLS';
+import TALENTS from 'common/TALENTS/deathknight';
+import { SpellLink } from 'interface';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { ApplyBuffEvent, SummonEvent } from 'parser/core/Events';
+import { When } from 'parser/core/ParseResults';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import { formatPercentage } from 'common/format';
+
+class CommanderOfTheDead extends Analyzer {
+  commanderBuffs = 0;
+  petSummons = 0;
+  petSummonIDs = [
+    SPELLS.MAGUS_SUMMON.id,
+    SPELLS.APOC_SUMMON.id,
+    SPELLS.ARMY_SUMMON.id,
+    TALENTS.SUMMON_GARGOYLE_TALENT.id,
+    SPELLS.DARK_ARBITER_TALENT_GLYPH.id,
+  ];
+  buffedPets: string[] = [];
+  summonedPets: string[] = [];
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(TALENTS.COMMANDER_OF_THE_DEAD_TALENT);
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.COMMANDER_OF_THE_DEAD_BUFF),
+      this.onBuffEvent,
+    );
+
+    this.addEventListener(Events.summon.by(SELECTED_PLAYER), this.onSummonEvent);
+  }
+
+  onBuffEvent(event: ApplyBuffEvent) {
+    let summonId = '';
+    if (event.targetInstance) {
+      summonId = event.targetID.toString() + '|' + event.targetInstance.toString(); // This is needed since the buff sometimes applies twice to the same summon.
+    } else {
+      summonId = event.targetID.toString();
+    }
+    if (!this.summonedPets.includes(summonId)) {
+      // This is the rare case of a pet being summoned without a summon event (potentially pre-combat).
+      this.summonedPets.push(summonId);
+      this.petSummons += 1;
+    }
+    if (!this.buffedPets.includes(summonId)) {
+      // Account for the case of double-buffing the same mob.
+      this.commanderBuffs += 1;
+      this.buffedPets.push(summonId);
+      console.log('Buffed!');
+      console.log(this.commanderBuffs);
```

Comment:
Console calls should be removed or disabled with `debug && console.log()` (where debug is a const at the top of the file set to false)

Reply:
I need to get better at looking for these before pushing...

Reply:
I did it just this last week at work, happens to everyone lol